src/infrastructure/transaction/CreateTransactionFromDTO.ts

```diff
@@ -349,20 +372,29 @@ const extractTransactionVersion = (version: number): number => {
  * @param recipient {string} Encoded hexadecimal recipient notation
  * @return {Address | NamespaceId}
  */
-const extractRecipient = (recipient: string): Address | NamespaceId => {
-    // If bit 0 of byte 0 is not set (like in 0x90), then it is a regular address.
-    // Else (e.g. 0x91) it represents a namespace id which starts at byte 1.
-    const bit0 = convert.hexToUint8(recipient.substr(1, 2))[0];
+const extractRecipient = (recipient: any): Address | NamespaceId => {
+    if (typeof recipient === 'string') {
+        // If bit 0 of byte 0 is not set (like in 0x90), then it is a regular address.
+        // Else (e.g. 0x91) it represents a namespace id which starts at byte 1.
+        const bit0 = convert.hexToUint8(recipient.substr(1, 2))[0];
```

Comment:
similar to `PlainMessage` when just toJSON method, it will create either a Address object  or NamespaceId object rather than hex value. Open for suggestion as well